File: packages/core/ios/ExpoKLineChartModule.swift
import ExpoModulesCore

public class ExpoKLineChartModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoKLineChart")

        View(KLineChartView.self) {
            // Legacy JSON mode (still supported)
            Prop("dataJson") { (view, json: String?) in
                guard let json = json else { return }
                view.setDataJson(json)
            }

            Prop("latestDataJson") { (view, json: String?) in
                guard let json = json else { return }
                view.setLatestDataJson(json)
            }

            // Native data source mode
            Prop("restBaseURL") { (view, value: String?) in
                view.setRestBaseURL(value ?? "")
            }

            Prop("wsBaseURL") { (view, value: String?) in
                view.setWsBaseURL(value ?? "")
            }

            Prop("symbol") { (view, value: String?) in
                view.setSymbol(value ?? "")
            }

            // Shared props
            Prop("configJson") { (view, json: String) in
                view.setConfigJson(json)
            }

            Prop("mainIndicator") { (view, value: String) in
                view.setMainIndicator(value)
            }

            Prop("subIndicators") { (view, value: [String]) in
                view.setSubIndicators(value)
            }

            Prop("timeFrame") { (view, value: String) in
                view.setTimeFrame(value)
            }

            Prop("locale") { (view, value: String?) in
                view.setLocale(value ?? "en")
            }

            Events("onCrosshairChange", "onCrosshairDismiss", "onVisibleRangeChange", "onContentHeightChange", "onFullscreenPress")
        }
    }
}

File: packages/expo-kline-chart/android/src/main/java/expo/modules/klinechart/ExpoKLineChartModule.kt
package expo.modules.klinechart

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoKLineChartModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoKLineChart")

        View(KLineChartView::class) {
            // Legacy JSON mode (still supported)
            Prop("dataJson") { view: KLineChartView, json: String? ->
                json?.let { view.setDataJson(it) }
            }

            Prop("latestDataJson") { view: KLineChartView, json: String? ->
                json?.let { view.setLatestDataJson(it) }
            }

            // Native data source mode
            Prop("restBaseURL") { view: KLineChartView, value: String? ->
                view.setRestBaseURL(value ?: "")
            }

            Prop("wsBaseURL") { view: KLineChartView, value: String? ->
                view.setWsBaseURL(value ?: "")
            }

            Prop("symbol") { view: KLineChartView, value: String? ->
                view.setSymbol(value ?: "")
            }

            // Shared props
            Prop("configJson") { view: KLineChartView, json: String ->
                view.setConfigJson(json)
            }

            Prop("mainIndicator") { view: KLineChartView, value: String ->
                view.setMainIndicator(value)
            }

            Prop("subIndicators") { view: KLineChartView, value: List<String> ->
                view.setSubIndicators(value)
            }

            Prop("timeFrame") { view: KLineChartView, value: String ->
                view.setTimeFrame(value)
            }

            Prop("locale") { view: KLineChartView, value: String? ->
                view.setLocale(value ?: "en")
            }

            Events("onCrosshairChange", "onCrosshairDismiss", "onVisibleRangeChange", "onContentHeightChange", "onFullscreenPress")
        }
    }
}
